### PR TITLE
Adding support for DC/OS service accounts.

### PR DESCRIPTION
**Pull request by TrevinTeacutter**

### CODE REVIEW

File: clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/DcosConfigurationProperties.groovy [hunk 1]
```diff
@@ -19,6 +19,8 @@ class DcosConfigurationProperties {
     String dcosUrl
     String user
     String password
+    String serviceId
```
Comment (TrevinTeacutter) [inline clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/DcosConfigurationProperties.groovy]: Should this just be folded into the user field?

Would make my validation logic easier.

File: clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/description/servergroup/DeployDcosServerGroupDescription.groovy [hunk 2]
```diff
@@ -19,7 +19,7 @@ class DeployDcosServerGroupDescription extends AbstractDcosCredentialsDescriptio
   Double disk
   Double gpus
   String constraints
-  List<String> fetch = new ArrayList<>()
+  List<Fetchable> fetch = new ArrayList<>()
```
Comment (willgorman) [inline clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/description/servergroup/DeployDcosServerGroupDescription.groovy:22]: What do you think about splitting the Fetchable stuff into a separate commit?

File: clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/DcosConfigurationProperties.groovy (hunk 1 above)
Comment (willgorman) [reply clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/DcosConfigurationProperties.groovy]: I'm ok with that.

File: clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/description/servergroup/DeployDcosServerGroupDescription.groovy (hunk 2 above)
Comment (TrevinTeacutter) [reply clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/description/servergroup/DeployDcosServerGroupDescription.groovy:22]: Separate commit or PR you mean? Cuz it is a separate commit right now :P

**Review by mtweten — APPROVED** [review]
+1

File: clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/description/servergroup/DeployDcosServerGroupDescription.groovy (hunk 2 above)
Comment (willgorman) [reply clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/description/servergroup/DeployDcosServerGroupDescription.groovy:22]: 😳

**Pull request merged**